src/layouts/EditPage.jsx

```diff
@@ -140,70 +167,46 @@ const EditPage = ({ match, isResourcePage, isCollectionPage, history, type }) =>
-        console.log(error)
-      }
-      
-      if (!content) return
+      if (_.isEmpty(pageData)) return
+
+      const {
+        pageContent,
+        pageSha,
+        netlifyTomlHeaderValues,
+        dirContent
+      } = pageData
+      if (!pageContent) return
       
-      try {
-        // split the markdown into front matter and content
-        const { frontMatter, mdBody } = frontMatterParser(Base64.decode(content));
-        
-        // retrieve CSP
-        const cspResp = await axios.get(`${process.env.REACT_APP_BACKEND_URL}/sites/${siteName}/netlify-toml`);
-        const { netlifyTomlHeaderValues } = cspResp.data;
-        const csp = new Policy(netlifyTomlHeaderValues['Content-Security-Policy']);
-
-        let leftNavPages
-        if (isCollectionPage) {
-          const collectionsApiEndpoint = getCollectionsApiEndpoint(apiEndpoint)
-          const collectionPagesResp = await axios.get(collectionsApiEndpoint);
-          const collectionResp = collectionPagesResp.data?.collectionPages;
-
-          // Retrieve third_nav_title from collection pages
-          leftNavPages = await Bluebird.map(collectionResp, async (collectionPage) => {
-            const collectionPageResp = await axios.get(`${collectionsApiEndpoint}/pages/${collectionPage.fileName}`)
-            const { content } = collectionPageResp.data;
-            const { frontMatter } = frontMatterParser(Base64.decode(content));
-            return {
-              ...collectionPage,
-              third_nav_title: frontMatter.third_nav_title,
-            }
-          });
-        }
+      const { frontMatter, mdBody } = frontMatterParser(Base64.decode(pageContent));
+      const csp = new Policy(netlifyTomlHeaderValues['Content-Security-Policy']);
+
+      let leftNavPages
+      if (isCollectionPage) {
+        const parsedFolderContents = parseDirectoryFile(dirContent)
+        leftNavPages = parsedFolderContents.map((name) => 
+          ({
+            fileName: name.includes('/') ? name.split('/')[1] : name,
+            third_nav_title: name.includes('/') ? name.split('/')[0] : null,
+          })
+        )
+      }
 
-        if (_isMounted) {
-          setCsp(csp)
-          setSha(sha)
-          setOriginalMdValue(mdBody.trim())
-          setEditorValue(mdBody.trim())
-          setFrontMatter(frontMatter)
-          setLeftNavPages(leftNavPages)
-          setIsLoadingPageContent(false)
-        }
-      } catch (err) {
-        errorToast(`There was a problem trying to load your page. ${DEFAULT_RETRY_MSG}`);
-        console.log(err);
+      if (_isMounted) {
+        setCsp(csp)
+        setSha(pageSha)
+        setOriginalMdValue(mdBody.trim())
+        setEditorValue(mdBody.trim())
+        setFrontMatter(frontMatter)
+        setLeftNavPages(leftNavPages)
+        setIsLoadingPageContent(false)
       }
     }
 
     loadPageDetails()
     return () => {
       _isMounted = false
     }
-  }, [])
+  }, [pageData])
 
   useEffect(() => {
```

Comment:
As with the above comment, not errors introduced in this PR but the `cspViolation` and `chunk` variables conflict with the state variables already declared above

Reply:
Resolved in https://github.com/isomerpages/isomercms-frontend/pull/379/commits/7d286dc115296f100690287fdb5c7b4a5da3e354